pkg/requests/requests.go

```diff
@@ -0,0 +1,78 @@
+package requests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+// ConstructGetURL creates a GET request from a base URL, a map of query
+// parameters, and a formatted path.
+func ConstructGetURL(
+	rawURL string, queries map[string][]string, format string, values ...interface{}) (*url.URL, error) {
+
+	urlPath, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+
+	urlPath.Path = fmt.Sprintf(format, values...)
+	var q url.Values = queries
+	urlPath.RawQuery = q.Encode()
+
+	return urlPath, nil
+}
```

Comment:
This package seems unnecessarily generic and in my opinion its usage bloats the code in `azuredevops.go`, especially with the `interface{}` map and formats. Would much rather see something like this:

```go
package requests

func NewGetRepositories(projectGroup, projectName string) (*url.URL, error) {}
func NewGetFile(projectGroup, projectName, filePath string) (*url.URL, error) {}
func NewGetProject(projectGroup, projectName) (*url.URL, error) {}
func NewGetProjects(projectGroup string) (*url.URL, error) {}
func NewGetGitRefs(projectGroup, projectName, refsFilter string) (*url.URL, error) {}
```

The above sample of functions would then be responsible for escaping all URL sections, adding the appropriate `apiVersion` query parameters, and so on.